### PR TITLE
Hostile mobs will set a "home" turf, and attempt to return to it upon losing aggro.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Mostly what it says on the tin, on init, hostile mobs will by default create a reference to the turf they natively spawned on, and will call that turf home.  If that hostile mob is drawn out of it's aggro zone, it will attempt to turn some of the distance  towards it's home to avoid from being constantly kited around and bated to new places.

Can be enabled/disabled on mobs by setting their create_home_turf variable to FALSE.

## Why It's Good For The Game

The ability to kite out lavaland mobs an infinite distance only makes sense if you're keeping in combat with those mobs. After a certain point, they should at least attempt to get back to their nest/hunting grounds/security post. This change adjusts this to a configurable option, in order to allow slightly more dynamic AIs, as well as keep the difficulty of extracting dangerous creatures out of their familiar environment real.

Willing to hear feedback on which critters shouldn't set a home as well, as I imagine not all of them should behave the same.
## Changelog
:cl:
add: Hostile mobs will attempt to return to their original position when losing their target.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 1]
```diff
@@ -3,60 +3,104 @@
...
+	var/attack_same = 0
+	///all range/attack/etc. calculations should be done from this atom, defaults to the mob itself, useful for Vehicles and such
+	var/atom/targets_from = null
+	///if true, equivalent to having a wanted_objects list containing ALL objects.
+	var/attack_all_objects = FALSE
+
+	///id for a timer to call LoseTarget(), used to stop mobs fixating on a target they can't reach
+	var/lose_patience_timer_id
+	///30 seconds by default, so there's no major changes to AI behaviour, beyond actually bailing if stuck forever
+	var/lose_patience_timeout = 30 SEONCDS
```
Comment (Couls) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:90]: oopsies

**Comment by necromanceranne** [comment]
What happens if a mob is brought to a new zlevel?

**Comment by Couls** [comment]
mentioned this on discord but I want to place it here for documentation purposes

I think this is a good idea for certain mobs that make sense, like syndicate simple mobs protecting their base or ruins

I don't think this is a good idea as a blanket setting on all hostiles because this punishes playing smart and trying to lure mobs away from where they used to be, or luring them in a high population area on station if you were an antag.

**Comment by ATH1909** [comment]
> mentioned this on discord but I want to place it here for documentation purposes
> 
> I think this is a good idea for certain mobs that make sense, like syndicate simple mobs protecting their base or ruins
> 
> I don't think this is a good idea as a blanket setting on all hostiles because this punishes playing smart and trying to lure mobs away from where they used to be, or luring them in a high population area on station if you were an antag.

my sentiments exactly

**Comment by Jared-Fogle** [comment]
Agree with Couls. How does this interact with antagonist miners trying to bring megafauna onto the station?

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm trying to wrap my head around the idea that not facetanking monsters so they can't hit you is considered smart or some implied advanced tactic. It's certainly not punishing a player, it just makes them have to approach the situation differently which is the point of the PR.

**Comment by ArcaneMusic** [comment]
Any suggestions on any other mobs that should/shouldn't set a home turf? SO far I have it culled down to just megafauna for relentless hunting, but I'm open to suggestions in the meantime. Otherwise, I'm happy with just leaving those as exceptions for now.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by ArcaneMusic** [comment]
I think I'll come back to this some other time, mobcode is a prison and I'd like to not get that locked in.

**Pull request closed without merging**